execute/aggregate.go

```diff
@@ -369,12 +421,53 @@ type aggregateState struct {
 	agg ValueFunc
 }
 
-func (t *simpleAggregateTransformation2) initializeState(chunk table.Chunk, current interface{}) ([]aggregateState, error) {
+// Recyclable represents an aggregate that can reuse state.
+// This is especially useful for transformations like quantile
+// which could potentially consume an inordinate amount of memory
+// if they didn't recycle resources.
+type Recyclable interface {
+	// Recycle should set an aggregate back to it's default state so that it
+	// can be used to calculate another result, and thereby prevent flux from
+	// having to allocate additional memory.
+	Recycle()
+
+	// Drop should decrement the amount of memory set aside for the aggregate.
+	// The value itself will be cleaned up by the garbage collector.
+	Drop()
+}
+
+func (a *aggregateState) Recycle() {
+	if agg, ok := a.agg.(Recyclable); ok {
+		agg.Recycle()
+	}
+}
+
+func (a *aggregateState) Drop() {
+	if agg, ok := a.agg.(Recyclable); ok {
+		agg.Drop()
+	}
+}
+
+type multiState []aggregateState
```

Comment:
Its not clear to me how multiState is used. Can you add a comment explaining its behavior?

Reply:
I could introduce a function that detects whether a multiState contains any non-recyclable items. Running a check on that before adding/removing states may resolve the problem?

Reply:
For reference, this commit contains an alternate fix: https://github.com/influxdata/flux/pull/4069/commits/daac803ee3223003a29ce68b8d5f3d06a36a8ef0